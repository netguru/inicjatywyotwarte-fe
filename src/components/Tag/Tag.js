import React from 'react'
import styled from 'styled-components'
import hexToRgba from 'hex-to-rgba'
import CloseIcon from '@material-ui/icons/Close'

const getBasicTagColor = theme => theme.tagColor

const getHoverBorderColor = theme => hexToRgba(getBasicTagColor(theme), '0.2')

const getBackgroundColor = theme => hexToRgba(theme.tagBackgroundColor)

const TagItem = styled.div`
  height: 2rem;
  color: ${props =>
    props.active ? props.theme.fontColorLight : getBasicTagColor(props.theme)};
  background: ${props => getBackgroundColor(props.theme)};
  box-sizing: border-box;
  border-radius: ${props => props.theme.borderRadiusBig};
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0 15px;
  margin: 4px 0;
  cursor: pointer;
  box-shadow: 0px 0px 0px 2px
    ${props => (props.active ? getBasicTagColor(props.theme) : 'transparent')};

  &:hover {
    box-shadow: 0px 0px 0px 2px ${props => getHoverBorderColor(props.theme)};
  }
`

const StyledCloseIcon = styled(CloseIcon)`
  margin-left: 0.5rem;
`

const Tag = ({ active, handleOnClick, tag, isTagsListOpen }) => {
  const shouldTagBeCollapsed = active || (!active && isTagsListOpen)

  return (
    <>
      {shouldTagBeCollapsed ? (
        <TagItem active={active} onClick={handleOnClick}>
          {tag}
          {active && <StyledCloseIcon fontSize='small' />}
        </TagItem>
      ) : null}
    </>
  )
}

export default Tag
