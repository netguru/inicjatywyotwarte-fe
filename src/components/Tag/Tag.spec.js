import React from 'react'
import Tag from './Tag'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Tag', () => {
  test('renders empty fragment when tag not active', () => {
    let wrapper = shallow(<Tag />);
    expect(wrapper.equals(<React.Fragment />)).toEqual(true);
  });

  test('renders TagItem when active', () => {
    let wrapper = shallow(<Tag active={true} />);
    expect(wrapper.find('TagItem').length).toEqual(1);
    expect(wrapper.find('Styled(CloseIcon)').length).toEqual(1);
  });

  test('match snapshot', () => {
    let wrapper = shallow(<Tag />);

    expect(toJson(wrapper)).toMatchSnapshot();

  });
});




