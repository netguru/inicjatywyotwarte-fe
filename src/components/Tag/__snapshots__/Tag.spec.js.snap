// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag match snapshot 1`] = `<Fragment />`;
